index.js

```diff
@@ -260,11 +260,15 @@ async function fastifyHttpProxy (fastify, opts) {
 
   fastify.register(From, fromOpts)
 
-  if (opts.proxyPayloads !== false) {
+  if (opts.preValidation === undefined && opts.proxyPayloads !== false) {
     fastify.addContentTypeParser('application/json', bodyParser)
     fastify.addContentTypeParser('*', bodyParser)
   }
 
+  if (opts.preValidation) {
+    fastify.addHook('preValidation', opts.preValidation)
+  }
+
```

Comment:
Why do we not proxy Payloads if preValidation is set?

Reply:
`req.body` is not parsed and comes as a stream if:

```
fastify.addContentTypeParser('application/json', bodyParser)
fastify.addContentTypeParser('*', bodyParser)
```

condition is executed, as described here: https://github.com/fastify/fastify-http-proxy/issues/86#issue-591840299

actually, the payload is passed in this case, see:

https://github.com/fastify/fastify-http-proxy/pull/312/files#diff-a561630bb56b82342bc66697aee2ad96efddcbc9d150665abd6fb7ecb7c0ab2fR269